pkg/commands/condition/create.go

```diff
@@ -0,0 +1,121 @@
+package condition
+
+import (
+	"io"
+
+	"github.com/fastly/cli/pkg/cmd"
+	"github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/global"
+	"github.com/fastly/cli/pkg/manifest"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/v8/fastly"
+)
+
+// CreateCommand calls the Fastly API to create an appropriate resource.
+type CreateCommand struct {
+	cmd.Base
+	manifest manifest.Data
+
+	// Required.
+	serviceVersion cmd.OptionalServiceVersion
+
+	// Optional.
+	autoClone     cmd.OptionalAutoClone
+	conditionType cmd.OptionalString
+	name          cmd.OptionalString
+	priority      cmd.OptionalInt
+	serviceName   cmd.OptionalServiceNameID
+	statement     cmd.OptionalString
+}
+
+// NewCreateCommand returns a usable command registered under the parent.
+func NewCreateCommand(parent cmd.Registerer, g *global.Data, m manifest.Data) *CreateCommand {
+	c := CreateCommand{
+		Base: cmd.Base{
+			Globals: g,
+		},
+		manifest: m,
+	}
+	c.CmdClause = parent.Command("create", "Create a condtion on a Fastly service version").Alias("add")
+
+	// Required flags
+	c.RegisterFlag(cmd.StringFlagOpts{
+		Name:        cmd.FlagVersionName,
+		Description: cmd.FlagVersionDesc,
+		Dst:         &c.serviceVersion.Value,
+		Required:    true,
+	})
+
+	// Optional flags
+	c.RegisterAutoCloneFlag(cmd.AutoCloneFlagOpts{
+		Action: c.autoClone.Set,
+		Dst:    &c.autoClone.Value,
+	})
+	c.CmdClause.Flag("name", "Condition name").Short('n').Action(c.name.Set).StringVar(&c.name.Value)
+	c.CmdClause.Flag("priority", "Condition priority").Action(c.priority.Set).IntVar(&c.priority.Value)
+	c.CmdClause.Flag("statement", "Condition statement").Action(c.statement.Set).StringVar(&c.statement.Value)
+	c.CmdClause.Flag("type", "Condition type").Action(c.conditionType.Set).StringVar(&c.conditionType.Value)
```

Comment:
We should use an enum type here as the data model https://developer.fastly.com/reference/api/vcl-services/condition/ for `type` suggests the values are constrained to `REQUEST`, `CACHE`, `RESPONSE`, `PREFETCH`.

See authtoken 'scope' for an example...

- https://github.com/fastly/cli/blob/main/pkg/commands/authtoken/create.go#L18-L20
- https://github.com/fastly/cli/blob/main/pkg/commands/authtoken/create.go#L49